eval.go

```diff
@@ -16,7 +16,7 @@ func (e constant) Eval() Func {
 }
 
 func (e variable) Eval() Func {
-	return func(args Arguments) float64 { return args[e.Name] }
+	return func(args Arguments) float64 { return args[e] }
```

Comment:
What would be the behavior when evaluating the following?
```go
x1 := Var("x")
x2 := Var("x")
f := Add(x1, x2)
```

And maybe more importantly, what behavior do we want. In my opinion we should have that `f = 2x`, because in the end both `x1` and `x2` refers to the symbol`x`. 

Why I'm asking this here is because I am a bit uncertain on how the hashing works in go. It is a similar uncertainty I feel regarding the comparison I commented on previously.

Reply:
I see, I think in this case we are comparing the elements not the pointers because there are no slices or maps in the variable struct. So, the behavior should be the expected one, I'll add a test

Reply:
This leads to some strange behavior, for instance if I say Argument{x1: 1, x2: 2}, which one should be used? Probably there should be a check that there is only one variable for each variable name.

Reply:
That is very true! Add it in the returning function in `Eval` and I'll resolve this comment and approve PR